### DISCUSSION TITLE
Coordinate Transformation Problem..

### DISCUSSION THREAD

**Discussion by Myzik-Dev** [iTwin/itwinjs-core#1329] [Q&A] [answered]
I got latitude and longitude using 'spatialToCartographic' function. 
After verification on Google Maps, the coordinates were well matched.

I changed the latitude and longitude to the ecef coordinate system for testing(EPSG:3857 WGS 84 to ECEF).

Then I converted it to a function 'ecefToSpatial' and marked its position, 
and it was more than 20km apart from its original position.

Was the way I tried wrong?

Please give me advice. Thank you.

( imodeljs ver : v2.12.1)

**Comment by kabentley** [comment]
Please see [this article](https://www.itwinjs.org/learning/geolocation). If you have a projected coordinate system, the ECEF transform only gives a rough approximation of location. It gets worse the farther you get from the ECEF origin. 

The only value of ECEF for iModels with a projected coordinate system is for getting an approximate location without requiring a round trip to a server (e.g. on mobile devices with GPS but no connectivity.)

**Comment by RBBentley** [comment, marked as answer]
If you do have a projected coordinate system and wish for a more accurate conversion from ECEF to spatial coordinates, try converting the ECEF to cartographic (Cartographic.fromEcef) and then iModel.cartographicToSpatial.   This will do the actual GCS projection and should give accurate results.   As Keith points out the ecef transform stored with projected iModels is an approximation.

**Comment by Myzik-Dev** [comment]
Thank you for your answer.

After changing the method, the difference from the original location was reduced to 5m.

Perhaps, with this result, it should be said that it has been resolved.

**Reply by kabentley** [reply]
I don't understand what you're trying to test. If you're using a projected coordinate system there is no need to use ECEF coordinates, that will only introduce errors. What exactly are you doing?

**Reply by Myzik-Dev** [reply]
It was a problem caused by testing by entering the wrong coordinate value.
The results were strange, so I would have tried a different method (ECEF).

As I said earlier, the wrong input value was a problem, and when I modified and applied the input value (WGS84), it was well converted.

In conclusion, the ECEF coordinate system was not used.
